channel/channel.go

```diff
@@ -13,20 +13,26 @@ import (
 	"github.com/statechannels/go-nitro/types"
 )
 
+type OnChainData struct {
+	LatestBlockNumber uint64 // the latest block number we've seen
+	Holdings          types.Funds
+	Outcome           outcome.Exit
+	StateHash         common.Hash
+}
+
+type OffChainData struct {
+	SignedStateForTurnNum       map[uint64]state.SignedState // Longer term, we should have a more efficient and smart mechanism to store states https://github.com/statechannels/go-nitro/issues/106
+	LatestSupportedStateTurnNum uint64                       // largest uint64 value reserved for "no supported state"
+}
+
 // Channel contains states and metadata and exposes convenience methods.
 type Channel struct {
+	state.FixedPart
 	Id      types.Destination
 	MyIndex uint
 
-	OnChainFunding    types.Funds
-	latestBlockNumber uint64 // the latest block number we've seen
-
-	state.FixedPart
-
-	SignedStateForTurnNum map[uint64]state.SignedState
-	// Longer term, we should have a more efficient and smart mechanism to store states https://github.com/statechannels/go-nitro/issues/106
-
-	latestSupportedStateTurnNum uint64 // largest uint64 value reserved for "no supported state"
+	OnChain  OnChainData
```

Comment:
We could consider using private members and exposing a `getter` function. This would prevent other structs from modifying `OnChain`, only calling `UpdateWithChainEvent` will do that.